tests/test_contracts.py

```diff
@@ -166,3 +166,55 @@ def test_search_invalid() -> None:
 
     msg = str(excinfo.value)
     assert 'all({n + m > 0 for n in numbers for m in numbers})' in msg
+
+
+class Player:
+    user: str
+
+
+class CPU(Player):
+    def __init__(self):
+        self.user = 'CPU'
+
+
+class HumanPlayer(Player):
+    def __init__(self):
+        self.user = 'Human'
+
+
+@check_contracts
+def _is_cpu(player: Player) -> bool:
+    return player.user == 'CPU'
+
+
+def test_type_not_instance_error() -> None:
+    """Test that the additional suggestion is added when the type of a class is passed in as an
+    argument instead of its instance
+
+    This test is coupled to the suggestion's arbitrarily chosen text, hence should be updated
+    when changing the suggestion text.
+    """
+
+    # Ternary assignment tricks the PyCharm linter into not recognizing the faulty argument
```

Comment:
The PyCharm linter doesn't actually matter for this test, since we're deliberately testing "bad" code. So it's actually better to make the test more explicit just by calling `_is_cpu(HumanPlayer)` in the test. And same below.